meter/tesla.go

```diff
@@ -101,10 +130,38 @@ func NewTesla(uri, usage string) (api.Meter, error) {
 	return decorateTesla(m, totalEnergy, batterySoC), nil
 }
 
+// Login calls login and saves the returned cookie
+func (m *Tesla) Login() error {
+	// username for the powerwall seems to always be customer; email is not required for authentication
+	payload := map[string]interface{}{"password": m.password, "username": "customer"}
+	payloadBytes, err := json.Marshal(payload)
+	if err != nil {
+		return err
+	}
+
+	// returns cookie which is saved in the cookie jar
+	resp, err := m.Client.Post(m.uri.String()+teslaLoginURI, "application/json", bytes.NewReader(payloadBytes))
+	if err != nil {
+		return fmt.Errorf("failed to post: %s", err)
+	}
+
+	defer resp.Body.Close()
+	// failed to login
+	if resp.StatusCode != http.StatusOK {
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return fmt.Errorf("failed to read response body: %s", err)
+		}
+		return fmt.Errorf("couldn't login: %s", body)
+	}
+
+	return nil
+}
+
 // CurrentPower implements the Meter.CurrentPower interface
 func (m *Tesla) CurrentPower() (float64, error) {
 	var res teslaMeterResponse
-	if err := m.GetJSON(m.uri+teslaMeterURI, &res); err != nil {
+	if err := m.GetJSON(m.uri.String()+teslaMeterURI, &res); err != nil {
```

Comment:
I'd prefer reverting. If not you would probably use something like https://stackoverflow.com/questions/34668012/combine-url-paths-with-path-join